backends/imgui_impl_vulkan.cpp

```diff
@@ -1052,13 +1068,19 @@ bool    ImGui_ImplVulkan_LoadFunctions(PFN_vkVoidFunction(*loader_func)(const ch
     IM_UNUSED(loader_func);
     IM_UNUSED(user_data);
 #endif
+
     g_FunctionsLoaded = true;
     return true;
 }
 
-bool    ImGui_ImplVulkan_Init(ImGui_ImplVulkan_InitInfo* info, VkRenderPass render_pass)
-{
-    IM_ASSERT(g_FunctionsLoaded && "Need to call ImGui_ImplVulkan_LoadFunctions() if IMGUI_IMPL_VULKAN_NO_PROTOTYPES or VK_NO_PROTOTYPES are set!");
+bool    ImGui_ImplVulkan_Init(ImGui_ImplVulkan_InitInfo* info, VkRenderPass render_pass) {
+	IM_ASSERT(g_FunctionsLoaded && "Need to call ImGui_ImplVulkan_LoadFunctions() if IMGUI_IMPL_VULKAN_NO_PROTOTYPES or VK_NO_PROTOTYPES are set!");
+
+	// Always laod the dynamic rendering function pointers, regardless of whether VK_NO_PROTOTYPES is defined
+	if (info->UseDynamicRendering) {
+		imgui_vkCmdBeginRenderingKHR = reinterpret_cast<PFN_vkCmdBeginRenderingKHR>(vkGetInstanceProcAddr(info->Instance, "vkCmdBeginRenderingKHR"));
+		imgui_vkCmdEndRenderingKHR = reinterpret_cast<PFN_vkCmdEndRenderingKHR>(vkGetInstanceProcAddr(info->Instance, "vkCmdEndRenderingKHR"));
+	}
```

Comment:
Thanks. Couple of suggestions here:

- In `ImGui_ImplVulkan_LoadFunctions()` you should use that function to load `imgui_vkCmdBeginRenderingKHR` and `imgui_vkCmdEndRenderingKHR` (if not available there we will have NULL and it is fine), otherwise you are bypassing the user loader in the combination of LoadFunctions+DynamicRendering. The case highlighted here should run only when LoadFunctions was not called.

- After the highlighted load, it is probably good to ASSERT that if UseDynamicRendering is true, then both functions are non-NULL.

Reply:
I think this should be addressed with eb95b8f.